lib/HttpClient.ts

```diff
@@ -243,8 +244,15 @@ export class HttpClient implements ifm.IHttpClient {
 
         let response: HttpClientResponse;
         while (numTries < maxTries) {
-            response = await this.requestRaw(info, data);
-
+            try{
+                response = await this.requestRaw(info, data);
+            }
+            catch (err) {
+                if(err && err.code && NetworkRetryErrors.indexOf(err.code) !== 1){
```

Comment:
Whoops, missed this in the initial review - this should be `if(err && err.code && NetworkRetryErrors.indexOf(err.code) > -1)`. As it is currently written it will retry for anything that isn't an ENOTFOUND

Reply:
Sorry about that